src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -291,6 +291,14 @@ def get_subscription(self, subscription=None):  # take id or name
     def get_subscription_id(self):
         return self.get_subscription()[_SUBSCRIPTION_ID]
 
+    # pylint: disable=no-self-use
+    def get_access_token_for_resource(self, username, tenant, resource):
+        tenant = tenant or 'common'
```

Comment:
this should the instance method, you should use the existing member of `self._creds_cache`